tofu/geom/_GG.pyx

```diff
@@ -2982,19 +2982,19 @@ def LOS_calc_signal(func, double[:,::1] ray_orig, double[:,::1] ray_vdir, res,
                                    num_threads)
             # ..................................................................
         if ani:
-            val_2d = func(pts, t=t, vect=-usbis, **fkwdargs)
+            val_arr = func(pts, t=t, vect=-usbis, **fkwdargs)
         else:
-            val_2d = func(pts, t=t, **fkwdargs)
+            val_arr = func(pts, t=t, **fkwdargs)
+        val_2d = val_arr
```

Comment:
Why is val_2d still necessary if we use val_arr now ?

Reply:
val_2d is a memory view. So computations and slicing in the other cases are much faster. val_arr is a numpy array, I thought this could be the origin of the errors in the unit tests (I tried different solutions).

I am benchmarking if there is a faster solution (between using only val_2d or using val_2d and val_arr).

Reply:
ok, thanks, I was not sure

Reply:
The difference is small but having the two variables seems to be faster.

Reply:
It looks like they take more or less the same time. I decided to only keep the memory view val_2d. I also tried to declare the variable to see if this helped, no visible changes. So I would leave the simplest solution (with only one variable)